packages/core/test/scp-cf/destination-accessor.spec.ts

```diff
@@ -1069,4 +1022,192 @@ describe('destination-accessor', () => {
+    }
+
+    function assertNothingCalledAndNullFound(
+      mocks: mocks,
+      destination: Destination | null
+    ) {
+      mocks.subscriberMocks.forEach(mock => expect(mock.isDone()).toBe(false));
+      mocks.providerMocks.forEach(mock => expect(mock.isDone()).toBe(false));
+      expect(destination).toBe(null);
+    }
+
+    describe('Combinations: subscriberUserToken x {alwaysSubscriber,alwaysProvider,subscriberFirst}', () => {
+      it('subscriberUserToken && alwaysSubscriberToken: should not send a request to retrieve remote provider destination and return subscriber destination.', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(
+          subscriberUserJwt,
+          alwaysSubscriber
+        );
+        expect(actual!.url).toBe(subscriberDestination.URL);
+
+        mocks.subscriberMocks.forEach(mock => expect(mock.isDone()).toBe(true));
+        mocks.providerMocks.forEach(mock => expect(mock.isDone()).toBe(false));
+      });
+
+      it('subscriberUserToken && alwaysProvider: should not sed a request to retrieve remote subscriber destination and return provider destination', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(
+          subscriberUserJwt,
+          alwaysProvider
+        );
+        assertSubscriberNotCalledAndProviderFound(mocks, actual!);
+      });
+
+      it('subscriberUserToken && subscriberFirst: should try subscriber first, provider second and return subscriber destination', async () => {
+        mockThingsForCombinations();
+
+        const requestSpy = jest.spyOn(
+          destinationService,
+          'fetchSubaccountDestinations'
+        );
+        const actualProvider = await fetchDestination(
+          subscriberUserJwt,
+          subscriberFirst
+        );
+
+        expect(requestSpy).toHaveBeenNthCalledWith(
+          1,
+          'https://destination.example.com',
+          subscriberServiceToken,
+          expect.anything()
+        );
+        expect(requestSpy).toHaveBeenNthCalledWith(
+          2,
+          'https://destination.example.com',
+          providerServiceToken,
+          expect.anything()
+        );
+        expect(actualProvider!.url).toBe(subscriberDestination.URL);
+      });
+    });
+
+    describe('providerUserToken x  {alwaysSubscriber,alwaysProvider,subscriberFirst}', () => {
+      it('providerUserToken x alwaysSubscriber: should return null since the token does not match subscriber', async () => {
+        const mocks = mockThingsForCombinations();
+
+        const actual = await fetchDestination(
+          providerUserJwt,
+          alwaysSubscriber
+        );
+        assertNothingCalledAndNullFound(mocks, actual);
+      });
+
+      it('providerUserToken x alwaysProvider: should not sed a request to retrieve remote subscriber destination and return provider destination.', async () => {
```

Comment:
```suggestion
      it('providerUserToken x alwaysProvider: should not send a request to retrieve remote subscriber destination and return provider destination.', async () => {
```
I think you copy&pasted this typo multiple times.